io/driver/module_unload_load.py

```diff
@@ -0,0 +1,140 @@
+        for line in genio.read_all_lines(config_path):
+            if module in line:
+                return line.split('=')[-1]
+
+    def flush_mpath(self, mdl):
+        """
+        flush the multipath
+        """
+        cmd = "lsmod | grep -i ^%s" % mdl
+        for i in range(20):
+            if process.getoutput(cmd).split(" ")[-1] == '0':
+                return True
+            process.system("multipath -F", ignore_status=True)
+            time.sleep(3)
+            continue
+        return False
+
+    def module_load_unload(self, module_list):
+        """
+        Unloading and loading the given module
+        """
+        for mod1 in module_list:
+            if linux_modules.module_is_loaded(mod1) is False:
+                linux_modules.load_module(mod1)
+                time.sleep(self.load_unload_sleep_time)
+
+        for _ in range(0, self.iteration):
+            for mdl in module_list:
+                sub_mod = self.get_depend_modules(mdl)
+                if sub_mod:
+                    for mod in sub_mod.split(' '):
+                        if mod == 'multipath':
+                            if self.flush_mpath(mdl) is False:
+                                self.error_modules.append(mdl)
+                            continue
+                        else:
+                            self.log.info("unloading sub module %s " % mod)
+                            process.system("rmmod %s" % mod)
+                self.log.info("unloading module %s " % mdl)
+                linux_modules.unload_module(mdl)
+                time.sleep(self.load_unload_sleep_time)
+                if linux_modules.module_is_loaded(mdl) is True:
+                    self.error_modules.append(mdl)
+                self.log.info("loading module : %s " % mdl)
+                linux_modules.load_module(mdl)
+                time.sleep(self.load_unload_sleep_time)
+                if linux_modules.module_is_loaded(mdl) is False:
+                    self.error_modules.append(mdl)
+
+    def test(self):
+        """
+        Begining the test here
+        """
+        pci_addrs = []
+        if self.module:
+            for modl in self.module:
+                self.mod_list.append(modl)
+        elif self.only_io is True:
+            for pci_addrs in pci.get_pci_addresses():
+                self.mod_list.append(pci.get_driver(pci_addrs))
+        else:
+            cmd = "find /lib/modules/%s/ -name /*.ko" % self.uname
+            for line in process.getoutput(cmd).splitlines():
+                driver = process.getoutput(line.split('/')[-1])
+                self.mod_list.append(driver.split('.')[0])
+
+        for mod in self.mod_list:
+            if self.built_in_module(mod) is True:
+                self.mod_list.remove(mod)
+        self.module_load_unload(self.mod_list)
+
+        if self.error_modules:
+            self.fail("few modules failed to load unload, please check logs")
```

Comment:
You can print those error modules here

Reply:
sure, I will add the print of error modules here.